include/maidsafe/crux/socket.hpp

```diff
@@ -60,8 +60,7 @@ class socket
         typename boost::asio::handler_type<CompletionToken,
                                            void(boost::system::error_code)>::type
         >::type
-    async_connect(const endpoint_type& remote_endpoint,
-                  CompletionToken&& token);
+    async_connect(endpoint_type remote_endpoint, CompletionToken&& token);
```

Comment:
Okay, I accept the patch as-is. We can always change the pass-by-const-ref later on.